### PR TITLE
Send type args for nullary method return types. + a little cleanup.

### PR DESCRIPTION
**Pull request by aemoncannon**
Previously you'd see a lot of "List", "Option" types showing up in the inspector without type arguments.

### CODE REVIEW

File: src/test/scala/org/ensime/core/RichPresentationCompilerSpec.scala [hunk 1]
```diff
@@ -208,6 +213,35 @@ class RichPresentationCompilerSpec extends FunSpec with Matchers with SLF4JLoggi
...
+    it("should show all type arguments in the inspector.") {
+      Helpers.withPosInCompiledSource(
+        "package com.example",
+        "class A { ",
+        "def banana(p: List[String]): List[String] = p",
+        "def pineapple: List[String] = List(\"spiky\")",
+        "}",
+        "object Main { def main { val my@@A = new A() }}") { (p, cc) =>
+          val info = cc.askInspectTypeAt(p).get
+          val sup = info.supers.find(sup => sup.tpe.name == "A").get;git
```
Comment (rorygraves) [inline src/test/scala/org/ensime/core/RichPresentationCompilerSpec.scala]: I don't think the word 'git' belongs here ;) - Compiler says no.

**Comment by aemoncannon** [comment]
Oops. Fixed.

**Comment by rorygraves** [comment]
@aemoncannon  The basic workflow text is broken.

http://fieldmarkeng.co.uk/job/Ensime-Server-CI-Build-PR-Builder/3418/consoleFull

The test is a little too specific - the arrow type id (see XXX in below) has changed - I suspect we dont care about the type id for ``` "(i: Int, s: String)Int" ``` so to fix you can simply replace the 133 one line 136 for ```BasicWorkflow.scala``` to _

```
Original
          case Some(SymbolInfo("testMethod", "testMethod", Some(OffsetSourcePosition(`fooFile`, 114)), ArrowTypeInfo("(i: Int, s: String)Int", XXX133XXX, BasicTypeInfo("Int", 1, 'class, "scala.Int", List(), List(), None, None), List(ParamSectionInfo(List((i, BasicTypeInfo("Int", 1, 'class, "scala.Int", List(), List(), None, None)), (s, BasicTypeInfo("String", 39, 'class, "java.lang.String", List(), List(), None, None))), false))), true, Some(_))) =>
Actual:
          case Some(SymbolInfo("testMethod", "testMethod" ,Some(OffsetSourcePosition(`fooFile`, 114)), ArrowTypeInfo("(i: Int, s: String)Int", XXX126XXX, BasicTypeInfo("Int", 1, 'class, "scala.Int", List(), List(), None, None), List(ParamSectionInfo(List((i, BasicTypeInfo("Int", 1, 'class, "scala.Int", List(), List(), None, None)), (s, BasicTypeInfo("String", 39, 'class, "java.lang.String", List(), List(), None, None))), false))),true,Some(_))) =>
```

**Comment by rorygraves** [comment]
@aemoncannon  Nice little fix - you are really motoring at the mo. :)

**Comment by fommil** [comment]
:smile: yup, and it's really appreciated. It was nice to come back to work with all the upgrades to the emacs client

**Comment by rorygraves** [comment]
And lets not forget @edani random fixes :smile:

**Comment by aemoncannon** [comment]
just pleased to have some hacking time lately :-)

**Comment by fommil** [comment]
:+1: @edani it's pretty fantastic to have such an active core contributor team. I must get my finger out and finish off my tickets.

**Comment by edani** [comment]
I have to admit, this is a pretty fantastic project to work on.

**Pull request merged**